src/main/kotlin/org/rust/cargo/runconfig/buildtool/CargoBuildContext.kt

```diff
@@ -49,6 +50,8 @@ class CargoBuildContext(
     var errors: Int = 0
     var warnings: Int = 0
 
+    val binaries: MutableList<Path> = mutableListOf()
```

Comment:
[not sure] I read the code around `CargoBuildContext` briefly, but probably we should synchronize or restrict access to this public mutable list?

Reply:
I have replaced `MutableList` with `List`, but I'm also not sure about synchronization.

Reply:
Now it looks a bit better. But anyway, `CargoBuildContext` provides a shared mutable state accessed from different threads (EDT, pooled threads, reader thread). This can cause problems in the future. Looks like this context can be separated into different ones. Probably like that (not sure, I haven't read the whole implementation):
* `errors` and `warning` are owned by `RsBuildEventsConverter`
* `indicator` and `processHandler` are owned by `CargoBuildManager`